### PR TITLE
NyxLex - Comparison test extension

### PR DESCRIPTION
**Pull request by NyxLex**

### CODE REVIEW

File: src/main/java/com/ss/ita/rozetka/pageobject/pages/ComparisonPage.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+public class ComparisonPage {
+    @Step("ComparisonPage: get product list")
+    private List<SelenideElement> getProductList() {
+        return $$x("//a[@class='product__heading']")
+                .shouldHave(CollectionCondition.sizeGreaterThan(0));
+    }
+
+    @Step("ComparisonPage: get product comparison list size")
+    public int sizeOfProductComparisonList() {
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/pageobject/pages/ComparisonPage.java:19]: rename

File: src/main/java/com/ss/ita/rozetka/pageobject/pages/ProductTypePage.java [hunk 2]
```diff
@@ -83,4 +86,24 @@ public ProductTypePage openProductsListPage(int numberProductsListPage) {
                 .map(price -> new BigDecimal(price))
                 .collect(Collectors.toList());
     }
+
+    @Step("ProductTypePage: get product comparison icons")
+    private List<SelenideElement> getProductComparisonIcons(int productCount) {
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/pageobject/pages/ProductTypePage.java:91]: this is used once, why separate method?

File: src/test/java/com/ss/ita/rozetka/test/ComparisonTest.java [hunk 3]
```diff
@@ -19,15 +19,19 @@ public void verifyAddingProductToComparison() {
...
                 .as("Notebooks subcategory page should be opened")
                 .isEqualTo("https://rozetka.com.ua/notebooks/c80004/");
-        Header header = productTypePage
-                .openProductPage(1)
-                .addProductToComparison();
 
-        assertThat(header.getProductsForComparisonCount())
-                .as("Count of comparison products should be equal to 1")
-                .isEqualTo(1);
+        int comparisonProductListSize = new Header()
```
Comment (toleksyn) [inline src/test/java/com/ss/ita/rozetka/test/ComparisonTest.java:28]: DO NOT create raw page objects in test
get it from existing PO

**Review by toleksyn — CHANGES_REQUESTED** [review]

**Review by NazarYakh — APPROVED** [review]

File: src/main/java/com/ss/ita/rozetka/pageobject/pages/ProductTypePage.java [hunk 4]
```diff
@@ -91,4 +92,18 @@ public ProductTypePage filterProductsByParameters(String parameter) {
...
+    @Step("ProductTypePage: add products to  comparison")
+    public ProductTypePage addNumberOfProductsToComparison(int productCount) {
+        List<SelenideElement> iconList = new ArrayList<>();
+        for (int i = 0; i < productCount; i++) {
+            SelenideElement icon = $x(format("(//button[@class='compare-button ng-star-inserted'])[%s]", i + 1));
+            iconList.add(icon);
+            iconList
+                    .get(i)
+                    .click();
+        }
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/pageobject/pages/ProductTypePage.java:105]: why don't just click the elem directly?

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/ss/ita/rozetka/pageobject/pages/ProductTypePage.java [hunk 5]
```diff
@@ -91,4 +92,18 @@ public ProductTypePage filterProductsByParameters(String parameter) {
 
         return new ProductTypePage();
     }
+
+    @Step("ProductTypePage: add products to  comparison")
+    public ProductTypePage addNumberOfProductsToComparison(int productCount) {
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/pageobject/pages/ProductTypePage.java:97]: fix step and name

**Pull request merged**